src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordApplicationHandler.java

```diff
@@ -156,6 +156,10 @@ public boolean canHandle(
                 if (!asylumCase.read(APPLICATION_WITHDRAW_EXISTS, String.class).isPresent()) {
                     asylumCase.write(APPLICATION_EDIT_LISTING_EXISTS, "Yes");
                 }
+            } else if (UPDATE_HEARING_REQUIREMENTS.toString().equalsIgnoreCase(applicationType)) {
+                if (!asylumCase.read(UPDATE_HEARING_REQUIREMENTS_EXISTS, String.class).isPresent()) {
+                    asylumCase.write(UPDATE_HEARING_REQUIREMENTS_EXISTS, "Yes");
+                }
             }
 
             asylumCase.write(DISABLE_OVERVIEW_PAGE, "Yes");
```

Comment:
I think clearing the flag needs to be handled for this event in an appropriate place. I think it is cleared specifically based on 'APPLICATION_EDIT_LISTING_EXISTS' flag which has no link to this event. Not sure to be handled as a part of this change or some follow up change. Just flagging.

Reply:
Clearing the flag will be in my ticket: RIA-2031. At some point we need to merge everything and deploy in one step to avoid broken links